File: bindings/lua/Polycode/Vector3.lua
class "Vector3"


function Vector3:__getvar(name)
	if name == "x" then
		return Polycode.Vector3_get_x(self.__ptr)
	elseif name == "y" then
		return Polycode.Vector3_get_y(self.__ptr)
	elseif name == "z" then
		return Polycode.Vector3_get_z(self.__ptr)
	end
end

function Vector3:__setvar(name,value)
	if name == "x" then
		Polycode.Vector3_set_x(self.__ptr, value)
		return true
	elseif name == "y" then
		Polycode.Vector3_set_y(self.__ptr, value)
		return true
	elseif name == "z" then
		Polycode.Vector3_set_z(self.__ptr, value)
		return true
	end
	return false
end
function Vector3:Vector3(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.Vector3(unpack(arg))
	end
end

function Vector3:set(x, y, z)
	local retVal = Polycode.Vector3_set(self.__ptr, x, y, z)
end

function Vector3:distance(vec2)
	local retVal = Polycode.Vector3_distance(self.__ptr, vec2.__ptr)
	return retVal
end

function Vector3:angleBetween(dest)
	local retVal = Polycode.Vector3_angleBetween(self.__ptr, dest.__ptr)
	return retVal
end

function Vector3:length()
	local retVal =  Polycode.Vector3_length(self.__ptr)
	return retVal
end

function Vector3:lengthSquared()
	local retVal =  Polycode.Vector3_lengthSquared(self.__ptr)
	return retVal
end

function Vector3:setLength(newLength)
	local retVal = Polycode.Vector3_setLength(self.__ptr, newLength)
	if retVal == nil then return nil end
	local __c = _G["Vector3"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function Vector3:dot(u)
	local retVal = Polycode.Vector3_dot(self.__ptr, u.__ptr)
	return retVal
end

function Vector3:crossProduct(vec2)
	local retVal = Polycode.Vector3_crossProduct(self.__ptr, vec2.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Vector3"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function Vector3:Normalize()
	local retVal =  Polycode.Vector3_Normalize(self.__ptr)
end

function Vector3:__delete()
	if self then Polycode.delete_Vector3(self.__ptr) end
end

File: bindings/javascript/Polycode/Vector3.js
function Vector3() {
	Object.defineProperties(this, {
		'x': { enumerable: true, configurable: true, get: Vector3.prototype.__get_x, set: Vector3.prototype.__set_x},
		'y': { enumerable: true, configurable: true, get: Vector3.prototype.__get_y, set: Vector3.prototype.__set_y},
		'z': { enumerable: true, configurable: true, get: Vector3.prototype.__get_z, set: Vector3.prototype.__set_z}
	})
}
Vector3.prototype.__get_x = function() {
	return Polycode.Vector3__get_x(this.__ptr)
}

Vector3.prototype.__set_x = function(val) {
	Polycode.Vector3__set_x(this.__ptr, val)
}

Vector3.prototype.__get_y = function() {
	return Polycode.Vector3__get_y(this.__ptr)
}

Vector3.prototype.__set_y = function(val) {
	Polycode.Vector3__set_y(this.__ptr, val)
}

Vector3.prototype.__get_z = function() {
	return Polycode.Vector3__get_z(this.__ptr)
}

Vector3.prototype.__set_z = function(val) {
	Polycode.Vector3__set_z(this.__ptr, val)
}


Vector3.prototype.set = function(x,y,z) {
	Polycode.Vector3_set(this.__ptr, x,y,z)
}

Vector3.prototype.distance = function(vec2) {
	return Polycode.Vector3_distance(this.__ptr, vec2)
}

Vector3.prototype.angleBetween = function(dest) {
	return Polycode.Vector3_angleBetween(this.__ptr, dest)
}

Vector3.prototype.length = function() {
	return Polycode.Vector3_length(this.__ptr)
}

Vector3.prototype.lengthSquared = function() {
	return Polycode.Vector3_lengthSquared(this.__ptr)
}

Vector3.prototype.setLength = function(newLength) {
	var retVal = new Vector3()
	retVal.__ptr = Polycode.Vector3_setLength(this.__ptr, newLength)
	return retVal
}

Vector3.prototype.dot = function(u) {
	return Polycode.Vector3_dot(this.__ptr, u)
}

Vector3.prototype.crossProduct = function(vec2) {
	var retVal = new Vector3()
	retVal.__ptr = Polycode.Vector3_crossProduct(this.__ptr, vec2)
	return retVal
}

Vector3.prototype.Normalize = function() {
	Polycode.Vector3_Normalize(this.__ptr)
}
